src/coprocessor/dag/expr/compare.rs

```diff
@@ -161,6 +161,70 @@ impl FnCall {
         do_coalesce(self, |v| v.eval_json(ctx, row))
     }
 
+    pub fn greatest_int(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        do_get_extremum(self, |e| e.eval_int(ctx, row), max)
+    }
+
+    pub fn greatest_real(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<f64>> {
+        do_get_extremum(self, |e| e.eval_real(ctx, row), |x, y| x.max(y))
+    }
+
+    pub fn greatest_decimal<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, Decimal>>> {
+        do_get_extremum(self, |e| e.eval_decimal(ctx, row), max)
+    }
+
+    pub fn greatest_time<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, Time>>> {
```

Comment:
Seems different with tidb's implementation? https://github.com/pingcap/tidb/blob/master/expression/builtin_compare.go#L552
@XuHuaiyu @zz-jason

Reply:
Hi, @bb7133 
In TiDB,  greatest_time returns type `string` which is not consistent with MySQL now,
but I think we'd better keep it be consistent with TiDB now to avoid unexpected error
when we try to push Greatest down to coprocessor.
This problem will be solved after https://github.com/pingcap/tidb/issues/6740 be fixed.

Reply:
Got it, I will fix this, thank you all!